scripts/v.dissolve/v.dissolve.py

```diff
@@ -33,36 +33,339 @@
+# % required: no
+# % options: univar,sql
+# %end
+# %rules
+# % requires_all: aggregate_method,aggregate_column
+# % requires_all: stats_column,aggregate_method,aggregate_column
+# %end
+
+"""Dissolve geometries and aggregate attribute values"""
 
-import os
 import atexit
+import json
+import subprocess
 
 import grass.script as grass
+
+# To use new style of import without changing old code.
+import grass.script as gs  # pylint: disable=reimported
 from grass.exceptions import CalledModuleError
 
 
-def cleanup():
-    nuldev = open(os.devnull, "w")
+# Methods supported by v.db.univar by default.
+UNIVAR_METHODS = [
+    "n",
+    "min",
+    "max",
+    "range",
+    "mean",
+    "mean_abs",
+    "variance",
+    "stddev",
+    "coeff_var",
+    "sum",
+]
+
+# Basic SQL aggregate function common between SQLite and PostgreSQL
+# (and the SQL standard) using their proper names and order from
+# their documentation.
+# Notably, this does not include SQLite total which returns zero
+# when all values are NULL.
+STANDARD_SQL_FUNCTIONS = ["avg", "count", "max", "min", "sum"]
+
+
+def get_methods_and_backend(methods, backend):
+    """Get methods and backed based on user-provided methods and backend"""
+    if methods:
+        if not backend:
+            in_univar = 0
+            neither_in_sql_nor_univar = 0
+            for method in methods:
+                if method not in STANDARD_SQL_FUNCTIONS:
+                    if method in UNIVAR_METHODS:
+                        in_univar += 1
+                    else:
+                        neither_in_sql_nor_univar += 1
+            # If all the non-basic functions are available in univar, use it.
+            if in_univar and not neither_in_sql_nor_univar:
+                backend = "univar"
+    elif backend == "sql":
+        methods = STANDARD_SQL_FUNCTIONS
+    elif backend == "univar":
+        methods = UNIVAR_METHODS
+    else:
+        # This is the default SQL functions but using the univar names (and order).
+        methods = ["n", "min", "max", "mean", "sum"]
+        backend = "sql"
+    if not backend:
+        backend = "sql"
+    return methods, backend
+
+
+def modify_methods_for_backend(methods, backend):
+    """Modify list of methods to fit the backend if they do not
+
+    This allows for support of the same method names for both backends.
+    It works both ways.
+    """
+    new_methods = []
+    if backend == "sql":
+        for method in methods:
+            if method == "n":
+                new_methods.append("count")
+            elif method == "mean":
+                new_methods.append("avg")
+            else:
+                new_methods.append(method)
+    elif backend == "univar":
+        for method in methods:
+            if method == "count":
+                new_methods.append("n")
+            elif method == "avg":
+                new_methods.append("mean")
+            else:
+                new_methods.append(method)
+    return new_methods
```

Comment:
Don't eliminate duplicate methods? `backend=sql methods=n,count` => `count,count`

Reply:
I didn't think of that, but you are right that may cause an issue.

With the new version, when result columns are explicitly specified, the user needs to include one method for every result column, i.e., duplicate methods are correct when they are for different columns.

Reply:
Or just return `set(new_methods)`?

Reply:
The `set(new_methods)` part is happening in _match_columns_and_methods_ but in a more complicated way because this is decided based on the other options. Specifically, it accommodates two basic use cases, when user wants a lot of results to be generated (e.g., exploratory analysis) then `set(new_methods)`-like operation is applied. When user wants finer control, no de-duplication is appropriate because then each method is applied to the corresponding column only.

The obvious criticism - which I myself have - is that it is complicated. However, I went through it again and the resulting interface should be simple enough without reading documentation covering both some experimenting in GUI and doing advanced scripting in Python. What is complicated is the implementation and the full documentation. (Error messages also guides you to using it right instead of just reporting what's wrong.)